toolsrc/src/vcpkg/commands.fetch.cpp

```
@@ -18,6 +20,36 @@ namespace vcpkg::Commands::Fetch
     {
         (void)args.parse_arguments(COMMAND_STRUCTURE);
         const std::string tool = args.command_arguments[0];
+
+        if (tool == "update")
+        {
+            const fs::path& git_exe = paths.get_tool_exe(Tools::GIT);
+            const std::string repo = "https://github.com/microsoft/vcpkg";
+            fs::path fetch_dir = paths.root / fs::u8path("vcpkg-fetch");
+
+            Files::Filesystem& fs = paths.get_filesystem();
+
+            fs.remove_all(fetch_dir, ignore_errors);
+            fs.create_directory(fetch_dir, VCPKG_LINE_INFO);
+
+            const std::string full_cmd = Strings::format(
+                R"(%s clone %s %s)", fs::u8string(git_exe), fs::u8string(repo), fs::u8string(fetch_dir));
+
+            auto output = System::cmd_execute_and_capture_output(full_cmd);
+
+            std::error_code ec;
+            fs.remove_all(paths.ports, ignore_errors);
```

I don't think ignore errors is the right behavior here. We want failures to be loud so we don't get "impossible" bugs where only some part of the tree got updated or similar.

It's also very strange to totally ignore these errors but then explicitly check for the git exit code on 48.